show/main.py

```diff
@@ -1550,35 +1554,14 @@ def show_run_snmp(db, ctx):
 # 'syslog' subcommand ("show runningconfiguration syslog")
 @runningconfiguration.command()
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-def syslog(verbose):
-    """Show Syslog running configuration
-    To match below cases(port is optional):
-    *.* @IPv4:port
-    *.* @@IPv4:port
-    *.* @[IPv4]:port
-    *.* @@[IPv4]:port
-    *.* @[IPv6]:port
-    *.* @@[IPv6]:port
-    """
-    syslog_servers = []
-    syslog_dict = {}
-    re_ipv4_1 = re.compile(r'^\*\.\* @{1,2}(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})(:\d+)?')
-    re_ipv4_2 = re.compile(r'^\*\.\* @{1,2}\[(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})\](:\d+)?')
-    re_ipv6 = re.compile(r'^\*\.\* @{1,2}\[([0-9a-fA-F:.]+)\](:\d+)?')
-    with open("/etc/rsyslog.conf") as syslog_file:
-        data = syslog_file.readlines()
-    for line in data:
-        if re_ipv4_1.match(line):
-            server =  re_ipv4_1.match(line).group(1)
-        elif re_ipv4_2.match(line):
-            server =  re_ipv4_2.match(line).group(1)
-        elif re_ipv6.match(line):
-            server =  re_ipv6.match(line).group(1)
-        else:
-            continue
-        syslog_servers.append("[{}]".format(server))
-    syslog_dict['Syslog Servers'] = syslog_servers
-    print(tabulate(syslog_dict, headers=list(syslog_dict.keys()), tablefmt="simple", stralign='left', missingval=""))
+@clicommon.pass_db
```

Comment:
Hi, thanks for the modification. I would suggest to read from `/etc/rsyslog.conf`
I have seen scenarios where the SYSLOG_SERVER is different between configDB and `rsyslog.conf`. 
For example, SYSLOG_SERVER have configured four servers while `rsyslogd` only contains a subset(2 or 3) of those server.
That could be caused by the fact that rsyslogd wasn't able to consume the configDB very well.

Reply:
@wen587 IMHO, getting syslog server info by directly grepping `/etc/rsyslog.conf` is a kind of overhead and simply looks like a bad design (configuration should be read from Config DB - that's a SONiC philosophy). Syslog configuration is by design to be rendered from a template file. If something is missing or a divergence is present - then it's completely user's responsibility.

Reply:
Hi @nazariig, Syslog server updating needs to restart rsyslog each time. It could be the restarting is too frequent and SONiC missed some updates. 
However, the `rsyslog.conf` if I understand it right, is the REAL number of syslog server starts.
We can treat ConfigDB as the ground truth if and only if SONiC consumes all the config correctly.
Hi @qiluo-msft , could you correct me if I am wrong?

Reply:
> Hi @nazariig, Syslog server updating needs to restart rsyslog each time. It could be the restarting is too frequent and SONiC missed some updates.

@wen587 i understand your concerns and you are totally right. This PR resolves the issue you mentioned

Reply:
@qiluo-msft can you please advise here?

Reply:
I checked many subcommands under `show running` and the implementation is not consistent. Reading `/etc/rsyslog.conf` has its own benefit, which acts more accurate as "running config". On the other side, reading ConfigDB acts more like "show user config".

@nazrrig Could you explain on the "overhead" part?

Reply:
@qiluo-msft IMHO, parsing generated files with regex for "running" config is an overhead. Still, if you believe that this is more reliable way - please let me know and i will align.

*P.S:* if we want to follow such an approach (instead of Config DB), would be great to define some kind of doc to explain where and how it supposed to be used.

Reply:
Please help explore parsing `/etc/rsyslog.conf` since this is the original solution. Let me know if you find it is unnecessary too complex.

Reply:
@qiluo-msft done